syntax = "proto3";

service NobelPrizeService {
  rpc GetLaureatesByCategoryAndYearRange (CategoryYearRequest) returns (LaureatesResponse);
  rpc GetLaureatesByKeyword (KeywordRequest) returns (LaureatesResponse);
  rpc GetLaureateByName (NameRequest) returns (LaureateDetailsResponse);
}

message CategoryYearRequest {
  string category = 1;
  int32 start_year = 2;
  int32 end_year = 3;
}

message KeywordRequest {
  string keyword = 1;
}

message NameRequest {
  string first_name = 1;
  string last_name = 2;
}

message LaureatesResponse {
  int32 total_laureates = 1;
}

message LaureateDetailsResponse {
  string year = 1;
  string category = 2;
  string motivation = 3;
}
